MH4U Dex/RegionEncyclopediaCollectionViewController.m

```diff
@@ -53,13 +56,34 @@ - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSe
 
 - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
 {
-    RegionEncylcopediaCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:NSStringFromClass([RegionEncylcopediaCollectionViewCell class]) forIndexPath:indexPath];
-    
-    // Configure the cell
-    cell.backgroundColor = [UIColor blueColor];
+    RegionEncyclopediaCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:NSStringFromClass([RegionEncyclopediaCollectionViewCell class]) forIndexPath:indexPath];
+    [cell displayContentsWithRegion:[self regionAtIndexPath:indexPath]];
     return cell;
 }
 
-#pragma mark <UICollectionViewDelegate>
+#pragma mark - Helper Methods
+
+- (Region *)regionAtIndexPath:(NSIndexPath *)indexPath
+{
+    return self.regions[indexPath.row];
+}
+
+#pragma mark - Navigation
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+    if ([segue.identifier isEqualToString:@"showRegionDetails"]) {
+        RegionContainerViewController *regionVC = (RegionContainerViewController *)segue.destinationViewController;
+        regionVC.managedObjectContext = self.managedObjectContext;
+        
+        if ([sender isMemberOfClass:[RegionEncyclopediaCollectionViewCell class]]) {
+            RegionEncyclopediaCollectionViewCell *cell = (RegionEncyclopediaCollectionViewCell *)sender;
+            Region *region = [self regionAtIndexPath:[self.collectionView indexPathForCell:cell]];
+            regionVC.regionName = region.name;
+            regionVC.regionKeyName = region.keyName;
+            regionVC.region = region;
```

Comment:
If you're setting a `region` property, are the `regionName` and `regionKeyName` properties on `regionVC` necessary?